ccm_web/server/src/canvas/canvas.scopes.ts

```diff
@@ -7,6 +7,7 @@ can be found in the API Developer Key interface in Canvas.
 
 const privilegeLevelOneScopes = [
   // Courses
+  'url:GET|/api/v1/courses/:id',
```

Comment:
I missed look at the API format of scopes and I ended up checking this `url:GET|/api/v1/courses/` scope from the API key console

I see that is there is an error due to miss matched scope then `query` object result is as below.
```
{
  error: "invalid_scope",
  error_description: "A requested scope is invalid, unknown, malformed, or exceeds the scope granted by the resource owner. The following scopes were requested, but not granted: url:GET|/api/v1/courses/:id",
}
```

I feel we need to check for `query.error` scenario and error out that will quickly fixes the configuration error which are initial to development and deployment. Currently the error in the log is this which is miss leading and make me think something wrong with session again.

`2021-06-15T18:42:48.277Z - canvas.controller.ts - warn - State variable returned from Canvas did not match session ID; throwing unauthorized exception...`

Add a case some thing like this in the `canvas.controller.ts`
```
if (query.error) {
      logger.info(`${error} ${error_description} `)
      throw new UnauthorizedException('Please verify your Canvas API configuration')
    }
```

Reply:
I'm a little confused. Can you provide steps to reproduce this?

Could this be a separate issue?

Reply:
Try checking these 2 scopes in CANVAS API KEY console and launch CCM, you will notice this error. You can deal this as separate issue. 
![Screen Shot 2021-06-15 at 3 13 03 PM](https://user-images.githubusercontent.com/8579775/122110158-39c32b80-cdec-11eb-8cee-b5c1b7a1ae99.png)

Reply:
So is this covered by #104?

Reply:
This is related to #104, but not the same. @pushyamig created a new issue (#129) and has an open PR.